Library/Homebrew/extend/os/linux/formula.rb

```diff
@@ -7,11 +7,38 @@ def shared_library(name, version = nil)
     "#{name}.so#{"." unless version.nil?}#{version}"
   end
 
+  undef allowed_missing_lib?
+  def allowed_missing_lib?(lib)
+    # lib:   Full path to the missing library
+    #        Ex.: /home/linuxbrew/.linuxbrew/lib/libsomething.so.1
+    # x -    Name of or a pattern for a library, linkage to which is allowed to be missing.
+    #        Ex. 1: "libONE.so.1"
+    #        Ex. 2: %r{(libONE|libTWO)\.so}
+    self.class.allowed_missing_libraries.any? do |x|
+      case x
+      when Regexp
+        x.match? lib
+      when String
+        lib.include? x
+      end
```

Comment:
We can add some logic here but... shall we wait and see if there is a need for that? Right now I suspect there will be 0 use cases for that error-catching "else".

Reply:
This will fail non-gracefully if there's a need for that. An `else` with a `raise` is minimal code to be maintained but will produce a nicer error message

Reply:
> This will fail non-gracefully if there's a need for that.

How would this fail? I thought it'll just quietly ignore it. I'm probably missing something...  Anyways, I agree with adding an `else` but shall it `odie` or `owarn`?

Reply:
Well, I guess "fail" is the wrong word here but "silently ignore the invalid non-String/non-Regex input" is probably a better one.